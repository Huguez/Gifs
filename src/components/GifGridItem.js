import React from 'react'
import { PropTypes } from 'prop-types';

export const GifGridItem = ( {imagen, title, id} ) => {

    // console.log( imagen );
    return (
        <div className="card animate__animated animate__fadeIn">
            <p> { title } </p>
            <img alt={title} src={ imagen } />
        </div>
    )
}

GifGridItem.prototype = {
    imagen: PropTypes.string.isRequired,
    title:  PropTypes.string.isRequired,
    id:     PropTypes.string
}
