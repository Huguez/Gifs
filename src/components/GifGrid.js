import React from "react";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import Proptypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data, loading } = useFetchGifs( category );
    
    return ( <>
        <h3> { category } </h3>
        { loading && <p className="animate__animated animate__flash "> Loading... </p> }

        <div className="card-grid"> 
            { data.map( ( item ) => { 
                return <GifGridItem key={item.id} {...item } />
            } ) }
        </div> 
    </>);
}

GifGrid.prototype  = {
    category: Proptypes.string.isRequired
}