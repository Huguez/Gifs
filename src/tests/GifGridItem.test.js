import "@testing-library/jest-dom";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';

import { GifGridItem } from '../components/GifGridItem'


configure({adapter: new Adapter()});
describe( 'Pruebas con GifGridItem.js', () => {

    const title = "El hacker"
    const url = "https://www.hackeo.com"
    const wrapper = shallow( <GifGridItem title={ title } imagen={ url } /> )

    test( 'debe de mostrar el componente correctamente', () => { 
        expect( wrapper ).toMatchSnapshot();
    } );

    test( 'debe de tener un parrafo en el title', () => {
        const p = wrapper.find( 'p' );
        expect( p.text().trim() ).toBe( title )
    } );

    test( 'debe de obtener la imagen igual al props ', () => {
        const img = wrapper.find( 'img' );
        
        expect( img.props().src ).toBe( url );
        
        expect( img.props().alt ).toBe( title );
        
    } );

    test( 'debe de tener animate__animated animate__fadeIn', () => {
        const div = wrapper.find( 'div' );

        // otra manera de hacerlo
        // expect( div.props().className ).toContain( "animate__animated animate__fadeIn" )
        expect( div.props().className.includes( "animate__animated animate__fadeIn" ) ).toBe( true )
    } );

    

} );