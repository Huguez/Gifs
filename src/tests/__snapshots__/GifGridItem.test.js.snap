// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con GifGridItem.js debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
     
    El hacker
     
  </p>
  <img
    alt="El hacker"
    src="https://www.hackeo.com"
  />
</div>
`;
